### ISSUE TITLE
Do not mark subscriptions as finished in-place

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9069]
Instead, retrieve a list of subscriptions that can be marked as finished and run the update in a different DAO/transaction. This will help us with the Rucio rules handling.

**Comment by ericvaandering** [comment]
Is this still planned?

**Comment by amaltaro** [comment]
Yep, we will try to get it in for the next WMAgent stable release.
